lighthouse-core/config/config.js

```diff
@@ -172,32 +199,73 @@ function merge(base, extension, overwriteArrays = false) {
     return merged;
   } else if (typeof extension === 'object') {
     if (typeof base !== 'object') throw new TypeError(`Expected object but got ${typeof base}`);
+    if (Array.isArray(base)) throw new TypeError('Expected object but got Array');
     Object.keys(extension).forEach(key => {
       const localOverwriteArrays = overwriteArrays ||
         (key === 'settings' && typeof base[key] === 'object');
-      base[key] = merge(base[key], extension[key], localOverwriteArrays);
+      base[key] = _merge(base[key], extension[key], localOverwriteArrays);
     });
     return base;
   }
 
   return extension;
 }
 
+/**
+ * Until support of jsdoc templates with constraints, type in config.d.ts.
+ * See https://github.com/Microsoft/TypeScript/issues/24283
+ * @type {LH.Config.Merge}
+ */
+const merge = _merge;
+
+/**
+ * @template T
+ * @param {Array<T>} array
+ * @return {Array<T>}
+ */
 function cloneArrayWithPluginSafety(array) {
   return array.map(item => {
-    return typeof item === 'object' ? Object.assign({}, item) : item;
+    if (typeof item === 'object') {
+      // Return copy of instance and prototype chain (in case item is instantiated class).
+      return Object.assign(
+        Object.create(
```

Comment:
did we need this? if we do, add a test for it?

Reply:
> did we need this? if we do, add a test for it?

we need it for supporting live gatherer instances or `{instance: liveGathererIntance, ...}`, which seems reasonable if we support live `implementation`s.

The reason I want it most is so `Config` is derived from `Config.Json`, but another approach to do that would be for shorthand expansion to drop `{instance: ...}`...we'd just need to make sure there was a `path` or `implementation` in that case.

I'll add a test but let me know if you'd rather just not support that

Reply:
that makes sense seems reasonable! just test case then :)